operator/validator/controller.go

```diff
@@ -835,7 +828,8 @@ func (c *controller) onShareInit(share *ssvtypes.SSVShare) (*validator.Validator
 	}
 
 	// Start a committee validator.
-	vc, found := c.validatorsMap.GetCommittee(operator.CommitteeID)
+	vc, found := c.validatorsMap.AddShareToCommittee(share)
```

Comment:
honestly this doesn't look atomic.
if there was no committee here, its not added, goes into `!found` there could be other share added to the same committee and it'll also go to `!found` then only one will be added and it will only have the latest share. 

though it might not be needed if this could never happen in parallel..

Reply:
Either make this atomic as well or make sure it can never go twice to this codepath and add a comment why you're not taking care of atomicity here.